server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -169,20 +180,46 @@ static Server makeJettyServer(DruidNode node, ServerConfig config)
     // to fire on main exit. Related bug: https://github.com/druid-io/druid/pull/1627
     server.addBean(new ScheduledExecutorScheduler("JettyScheduler", true), true);
 
-    ServerConnector connector = new ServerConnector(server);
-    connector.setPort(node.getPort());
-    connector.setIdleTimeout(Ints.checkedCast(config.getMaxIdleTime().toStandardDuration().getMillis()));
-    // workaround suggested in -
-    // https://bugs.eclipse.org/bugs/show_bug.cgi?id=435322#c66 for jetty half open connection issues during failovers
-    connector.setAcceptorPriorityDelta(-1);
+    final ServerConnector[] connectors = new ServerConnector[config.isPlainText() && config.isTls() ? 2 : 1];
+    if (config.isPlainText()) {
+      log.info("Creating http connector with port [%d]", node.getPlainTextPort());
+      connectors[0] = new ServerConnector(server);
+      connectors[0].setPort(node.getPlainTextPort());
+    }
+    if (config.isTls()) {
+      log.info("Creating https connector with port [%d]", node.getTlsPort());
+      final SslContextFactory sslContextFactory = new SslContextFactory(tlsServerConfig.getKeyStorePath());
+      sslContextFactory.setKeyStoreType(tlsServerConfig.getKeyStoreType());
+      sslContextFactory.setKeyStorePassword(tlsServerConfig.getKeyStorePassword());
+      sslContextFactory.setCertAlias(tlsServerConfig.getCertAlias());
+      sslContextFactory.setKeyManagerPassword(tlsServerConfig.getKeyManagerPassword());
+      // setup https connector
+      final HttpConfiguration httpsConfiguration = new HttpConfiguration();
+      httpsConfiguration.setSecureScheme("https");
+      httpsConfiguration.setSecurePort(node.getTlsPort());
+      httpsConfiguration.addCustomizer(new SecureRequestCustomizer());
+      connectors[1] = new ServerConnector(
```

Comment:
if `config.isPlainText()` is false but `config.isTls()` is true, connectors[] would have size 1 and this array access would be out of bounds